src/controllers/portfolio/portfolio.ts

```diff
@@ -664,6 +720,87 @@ export class PortfolioController extends EventEmitter {
     this.emitUpdate()
   }
 
+  /**
+   * Learn new tokens from humanizer and debug_traceCall
+   */
+  async learnTokens(tokens: string[], networkId: NetworkId, accountId: AccountId) {
+    const storagePreviousHints = await this.#storage.get('previousHints', {})
+
+    const learnedTokens = storagePreviousHints.learnedTokens || {}
+    const networkLearnedTokens = learnedTokens[networkId] || {}
+
+    const network = this.#settings.networks.find((x) => x.id === networkId)
+
+    if (!network) throw new Error('network not found')
+
+    const portfolioLib = this.initializePortfolioLibIfNeeded(accountId, networkId, network)
+
+    // Get the limit for the network
+    const limit = portfolioLib.deploylessTokens.isLimitedAt24kbData
```

Comment:
this limit is not the same. This effectively means that we limit the learned tokens to one page of the deployless token mechanism, which makes sense to a big extent but at the same time not really because the RPC might change the limit, which is a bit chaotic

I recommend we just use a fixed number for this, eg 50